### PR TITLE
Issue #13999: Resolve pitest suppression for validateDefaultJavadocTokens method of AbstractJavadocCheck

### PR DESCRIPTION
**Pull request by TanayMorakhia**
part of #13999 

Explanation - All the Overridden implementation of the method `getRequiredJavadocTokens()` are of length more than 0.

Diff Regression config: https://gist.githubusercontent.com/TanayMorakhia/431cb93bf15c238967db9914a3d164ec/raw/575ff26646e5f029e9f9053a8878f27859bb186b/AbstractJavaDocCheck.xml

Diff Regression projects: https://gist.githubusercontent.com/TanayMorakhia/f3a248e4d705e5b3e0defe44c8bc8452/raw/6a2871822b807d93bcb58a449f79d94a55702f35/project.properties

### CODE REVIEW

**Comment by TanayMorakhia** [comment]
Github, generate report

File: src/main/java/com/puppycrawl/tools/checkstyle/checks/javadoc/AbstractJavadocCheck.java [hunk 1]
```diff
@@ -217,18 +217,16 @@ public void init() {
      * @throws IllegalStateException when validation of default javadoc tokens fails
      */
     private void validateDefaultJavadocTokens() {
-        if (getRequiredJavadocTokens().length != 0) {
```
Comment (romani) [inline src/main/java/com/puppycrawl/tools/checkstyle/checks/javadoc/AbstractJavadocCheck.java:220]: to kill mutation you need to find javadoc check that has no required tokens and use it
or make test Check that has no required tokens and execute some validation in test.
We have several Checks that are defined in Test class as inner class, so just search in sources for example.

**Review by romani — CHANGES_REQUESTED** [review]
regression will unlikely to find anything, as it is validation of configuration of Check 
not a javadoc content.


items:

File: src/main/java/com/puppycrawl/tools/checkstyle/checks/javadoc/AbstractJavadocCheck.java (hunk 1 above)
Comment (TanayMorakhia) [reply src/main/java/com/puppycrawl/tools/checkstyle/checks/javadoc/AbstractJavadocCheck.java:220]: Ok... was occupied for sometime, will start working on this now.

File: src/main/java/com/puppycrawl/tools/checkstyle/checks/javadoc/AbstractJavadocCheck.java [hunk 2]
```diff
@@ -217,18 +217,16 @@ public void init() {
...
+        for (final int javadocToken : getRequiredJavadocTokens()) {
+            if (Arrays.binarySearch(defaultJavadocTokens, javadocToken) < 0) {
+                final String message = String.format(Locale.ROOT,
+                        "Javadoc Token \"%s\" from required javadoc "
+                            + "tokens was not found in default "
+                            + "javadoc tokens list in check %s",
+                        javadocToken, getClass().getName());
+                throw new IllegalStateException(message);
             }
         }
```
Comment (nrmancuso) [inline src/main/java/com/puppycrawl/tools/checkstyle/checks/javadoc/AbstractJavadocCheck.java:231]: The existing code in this method is crazy, we need to clean this up. Binary search and sorting of default tokens is overkill for such a simple task. We can just do something like this here:
```suggestion
        final Set<Integer> defaultTokens = Arrays.stream(getDefaultJavadocTokens())
                .boxed()
                .collect(Collectors.toUnmodifiableSet());

        final List<Integer> missingRequiredTokens = Arrays.stream(getRequiredJavadocTokens())
                .boxed()
                .filter(token -> !defaultTokens.contains(token))
                .collect(Collectors.toUnmodifiableList());

        missingRequiredTokens.forEach(token -> {
            final String message = String.format(Locale.ROOT,
                    "Javadoc Token \"%s\" from required javadoc"
                            + "tokens was not found in default "
                        + "javadoc tokens list in check %s",
                    JavadocUtil.getTokenName(token), getClass().getName());
            throw new IllegalStateException(message);
        });
```

IMO this is much easier to read and understand, provides a much more helpful exception message, avoids mutable collections, and should kill the mutation.

**Comment by nrmancuso** [comment]
@TanayMorakhia let's get this out of draft mode and across the finish line

**Comment by nrmancuso** [comment]
@TanayMorakhia please make error prone happy, and patch checker suppression file

**Comment by TanayMorakhia** [comment]
@nrmancuso I added suppression for 
https://github.com/checkstyle/checkstyle/actions/runs/8194664368/job/22411239165?pr=14477#step:6:651

still getting the same in unnecessary suppression
https://github.com/checkstyle/checkstyle/actions/runs/8194664368/job/22411239404?pr=14477#step:6:949

**Comment by nrmancuso** [comment]
> @nrmancuso I added suppression for https://github.com/checkstyle/checkstyle/actions/runs/8194664368/job/22411239165?pr=14477#step:6:651
> 
> still getting the same in unnecessary suppression https://github.com/checkstyle/checkstyle/actions/runs/8194664368/job/22411239404?pr=14477#step:6:949

Did you download the archive, unzip it, and apply via `git apply`?

**Review by nrmancuso — APPROVED** [review]

**Comment by nrmancuso** [comment]
@romani please complete review

**Review by romani — APPROVED** [review]
ok to merge

**Pull request merged**